scripts/synthetix/facilitate_trade.js

```diff
@@ -73,44 +88,89 @@ module.exports = async (callback) => {
     const exchangeRate = await snxjs.ExchangeRates.rateForCurrency(sETHKey)
     const formatedRate = snxjs.utils.formatEther(exchangeRate)
-    console.log("sETH Price", snxjs.utils.formatEther(exchangeRate))
+    console.log("Oracle sETH Price (in sUSD)", formatedRate)
+
+    const exchangeBuyETHPrice = await estimatePrice(
+      sETH.exchangeId,
+      sUSD.exchangeId,
+      floatToErc20Units(MIN_SELL_USD, sUSD.decimals),
+      networkId
+    )
+    console.log("Exchange buy  sETH price (in sUSD)", 1 / exchangeBuyETHPrice)
+
+    const minSellETH = floatToErc20Units(MIN_SELL_USD / formatedRate, sETH.decimals)
+    const exchangeSellETHPrice = await estimatePrice(sUSD.exchangeId, sETH.exchangeId, minSellETH, networkId)
+    console.log("Exchange sell sETH price (in sUSD)", exchangeSellETHPrice)
 
     // Using synthetix's fees, and formatting their return values with their tools, plus parseFloat.
     const sETHTosUSDFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sETHKey, sUSDKey)))
     const sUSDTosETHFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sUSDKey, sETHKey)))
 
-    // Compute buy-sell amounts based on unlimited orders with rates from above.
-    const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
-      1 / (formatedRate * (1 - sUSDTosETHFee)),
-      sETH.decimals,
-      sUSD.decimals
-    )
-    const { base: sellETHAmount, quote: buySUSDAmount } = getUnlimitedOrderAmounts(
-      formatedRate * (1 + sETHTosUSDFee),
-      sUSD.decimals,
-      sETH.decimals
-    )
-
-    const buyAmounts = [buyETHAmount, buySUSDAmount]
-    const sellAmounts = [sellSUSDAmount, sellETHAmount]
-
-    // Fetch auction index and declare validity interval for orders.
-    // Note that order validity interval is inclusive on both sides.
-    const batchId = (await exchange.getCurrentBatchId.call()).toNumber()
-    const validFroms = Array(2).fill(batchId)
-    const validTos = Array(2).fill(batchId)
-
-    // Avoid querying exchange by tokenAddress for fixed tokenId
-    const buyTokens = [sETH, sUSD].map((token) => token.exchangeId)
-    const sellTokens = [sUSD, sETH].map((token) => token.exchangeId)
-
-    const gasPrices = await (await fetch(gasStationURL[networkId])).json()
-    const scaledGasPrice = parseInt(gasPrices[argv.gasPrice] * argv.gasPriceScale)
-    console.log(`Using current "${argv.gasPrice}" gas price scaled by ${argv.gasPriceScale}: ${scaledGasPrice}`)
-    await exchange.placeValidFromOrders(buyTokens, sellTokens, validFroms, validTos, buyAmounts, sellAmounts, {
-      from: account,
-      gasPrice: scaledGasPrice,
-    })
+    // Initialize order array.
+    const orders = []
+
+    // Compute buy-sell amounts based on unlimited orders with rates from above when the price is right.
+    const ourBuyETHRate = formatedRate * (1 - sUSDTosETHFee)
+    if (ourBuyETHRate > 1 / exchangeBuyETHPrice) {
```

Comment:
When I first reviewed this I was also mostly focussing on the print statements which convinced me the inequality was correct. The fact that we are inverting the value in the print statement and have to remember to do it here as well is a bit confusing. Maybe we can make it clearer by naming the variable `exchangeBuyETHPriceInEth` so that the unit of measurement is clearer.

In natural language, I'd agree with @fedgiac that we want to place an order if "snxBuyEthRateInSusd" is larger than the "gpSellEthRateInSusd" that is the synthetix exchange is willing to buy eth at a price higher than GP is willing to sell it for (meaning we have a match). However since we are placing sell orders we would have to place a sell sUSD order.

So maybe the logic should be `if snxSellSusdRateInEth < gpBuySusdRateInEth` (SNXchange is willing to sell for less than GP is willing to buy for), place a sell SUSD order.

Reply:
I would be happy and willing to change the internal variable names, but I would like to keep the print statements in a human readable prices (i.e. the price of ETH in USD so that the user has something to relate to). I hope that I am never able to interpret the price of USD in ETH unless they eventually become one-to-one! At the time of writing this, the price of USD in ETH is at 0.004112518506